packages/discovery-react-components/src/components/SearchResults/components/ResultElement/ResultElement.tsx

```diff
@@ -71,9 +76,27 @@ export const ResultElement: React.FunctionComponent<ResultElementProps> = ({
 
   const elementBodyProps = {
     className: elementBodyClassNames.join(' '),
-    'data-testid': `search-result-element-body-${elementType}`
+    'data-testid': `search-result-element-body-${elementType}`,
+    'aria-label': elementLabel,
+    ...(elementType === 'table' ? { role: 'img' } : {})
   };
 
+  useLayoutEffect(() => {
+    if (elementType === 'table') {
+      DOMPurify.addHook('afterSanitizeAttributes', function(node) {
+        if (node.tagName === 'TABLE') {
+          node.setAttribute('role', 'presentation');
+        }
+      });
+    }
+
+    return () => {
+      if (elementType === 'table') {
+        DOMPurify.removeHook('afterSanitizeAttributes');
+      }
+    };
+  }, [elementType]);
```

Comment:
1. This is duplicated in multiple places. Better to create a custom hook with the main logic.
2. This should also have a good comment on why the code is needed, what it is doing.
3. Why do we need the `removeHook`?